src/instructlab/config/init.py

```diff
@@ -314,6 +314,19 @@ def walk_and_print_system_profiles(
     return cfg
 
 
+def is_hpu_available() -> bool:
+    # Third Party
```

Comment:
Could you add a docstring here?

Reply:
Also, should this be a `util`?

Reply:
Should have a unit tests for this new func as well

Reply:
I do not think this should be a utility since it is directly used for auto-detection code, which also all lives in here, moving all of that is a different discussion